rider-fsharp/src/main/java/com/jetbrains/rider/plugins/fsharp/editorActions/FSharpEnterHandlerDelegate.kt

```diff
@@ -434,13 +496,224 @@ class FSharpEnterHandlerDelegate : EnterHandlerDelegateAdapter() {
     return true
   }
 
+  private fun isFirstTokenOnLine(editor: Editor, offset: Int): Boolean {
+    val iterator = editor.highlighter.createIterator(offset)
+    iterator.retreat()
+    while (!iterator.atEnd() && isIgnored(iterator.tokenType)) {
+      iterator.retreat()
+    }
+    return iterator.atEnd() || iterator.tokenType == FSharpTokenType.NEW_LINE
+  }
+
+  private fun isLastTokenOnLine(editor: Editor, offset: Int): Boolean {
+    val iterator = editor.highlighter.createIterator(offset)
+    iterator.advance()
+    while (!iterator.atEnd() && isIgnored(iterator.tokenType)) {
+      iterator.advance()
+    }
+    return iterator.atEnd() || iterator.tokenType == FSharpTokenType.NEW_LINE
+  }
+
+  private fun getLineIndent(editor: Editor, line: Int): LineIndent? {
```

Comment:
I do wonder if there are any similar APIs in the platform already? It seems it could be useful in languages like Python, for example

Reply:
I searched for a while but didn't find it. In addition, it seems that there are no intermediate comment tokens in Python like this:
```    (* comment *) code```